### ISSUE TITLE
Drop message in favour of a comment

### ISSUE THREAD

**Issue by rgaiacs** [sdruskat/citation-file-format#9]

**Comment by sdruskat** [comment]
What would the rationale for this be? I've included the `message` in the data explicitly because it might contain specific instructions to users of the file, e.g., "If you cite use of the software, cite the software itself; if you cite the existence of the software in the context of this and that framework, cite the paper", etc.

I thought that this would be something that actors may want to reproduce automatically, e.g., in log outputs, etc.

**Comment by rgaiacs** [comment]
I thought that you included `message` only to make it more human readable, in other words, to help users know what the file was about.

> "If you cite use of the software, cite the software itself; if you cite the existence of the software in the context of this and that framework, cite the paper"

I would suggest to replace the single mandatory top-level `message` field with a optional `message` field per item. This will help users that have many papers with many conditions. For example, a small group could use

~~~
- type: software-code
  authors:
    - name: Druskat::Stephan
      orcid: 0000-0003-4925-7248
  title: Stephan's Research Software
  doi: 10043/zenodo.1234
~~~

but a big group could use

~~~
- message: ""If you cite use of the software"
  type: software-code
  authors:
    - name: Druskat::Stephan
      orcid: 0000-0003-4925-7248
  title: Stephan's Research Software
  doi: 10043/zenodo.1234
- message: "If you cite the existence of the software in the context of this and that framework"
  type: paper
  authors:
    - name: Druskat::Stephan
      orcid: 0000-0003-4925-7248
  title: Stephan's Research Software Paper
  doi: 10043/zenodo.5678
~~~

**Comment by sdruskat** [comment]
I see. I had actually thought about including a message field per reference but dropped it in favour of the single mandatory field. 

I think that for where we are with software citation in the grand scheme of things it is necessary to include such a message in order to enforce instructions (even if it is only the minimal "If you use ... please cite ..."). Hence mandatory. If someone stumbles across the file without knowing what it is it's much easier to ignore it (or take it for a random reference file or similar) without the message. 

Also I originally made it a global field for a set of references in order to enforce creation of a single file for a single "piece of software" (however defined). Having one message per reference IMHO suggests that it its okay to use one "global" file for more than one "piece of software", which would not be SoC, and impede discoverability.

Does this make sense?

All in all I don't see the downside of having a mandatory message as it can always be ignored by downstream tools (or be set to `null`). If you want to enable reference-specific instructions within a reference - which is a good idea I guess - should we create a new field `comment`, `instruction` or `use case`, or specify `notes` to be the field used for this?

**Comment by rgaiacs** [comment]
> I think that for where we are with software citation in the grand scheme of things it is necessary to include such a message in order to enforce instructions (even if it is only the minimal "If you use ... please cite ..."). Hence mandatory. If someone stumbles across the file without knowing what it is it's much easier to ignore it (or take it for a random reference file or similar) without the message.

I understand why you want to have the message and that's why I suggested use comments for it.

Another reason why I wanted to make it a comment is for semantic. YAML requires that the first level be a list or a dictionary but you can't mix it (AFAIK), in other words, you need to have

~~~
- a
- b
- c
~~~

or

~~~
a: A
b: B
c: C
~~~

but you can't have

~~~
a: A
- b
c: C
~~~

Because of the list concept, would be nice that all the elements on the list are works to be cited and the `message` one isn't.

> Also I originally made it a global field for a set of references in order to enforce creation of a single file for a single "piece of software" (however defined). Having one message per reference IMHO suggests that it its okay to use one "global" file for more than one "piece of software", which would not be SoC, and impede discoverability.

I didn't understand this part.

> should we create a new field `comment`, `instruction` or `use-case`, or specify `notes` to be the field used for this?

I don't have strong preference for the key.

**Comment by sdruskat** [comment]
## Re **semantics of YAML requiring a top-level list or dictionary**:

I think this is resolved through fixing #7. 

The top-level structure will be a *dictionary* with two keys: `message` and `references`. `references`' value is a list of references (for terms "reference" v "work" see follow-up discussion in https://github.com/sdruskat/citation-file-format/pull/13/commits/7175376c062f424b085587ba14a389a303a299f3).

## To clarify **global message vs reference-specific message**:

I'd rather have one (mandatory, see above, comment v item) message per file, as each discrete "piece of software" (package, project, application, script, etc.) should have its own file. 

If you allow one message per *reference*, this would suggest that a single CITATION.cff file can contain references for more than one "piece of software". I think that the latter should be discouraged, as it would 

1. lessen discoverability (i.e., it will be harder to find the specific reference for a specific software), and
2. it breaks the *Separation of Concerns* principle. 

*However*, in some cases it might not be feasible/possible to have separate CITATION.cff files for different "pieces of software" within one project - perhaps due to the structure of the repository: E.g., one Java package (and therefore, one folder in the repo structure) contains classes that each implement an algorithm which should be made citable via CFF. 

For such a case I would introduce another key on the *reference* level. My suggestions in https://github.com/sdruskat/citation-file-format/issues/9#issuecomment-332863996 were: 

1. a separate key, or 
2. re-use the `notes` key. 

I've since come to the conclusion that it would be best to have a separate key for this purpose, again due to Separation of Concerns, as the `notes` key may be used for more general purpose notes. I suggest to use ``, but I'd like to hear other opinions on this in a separate issue.
